endpoints/openrtb2/auction.go

```diff
@@ -1507,6 +1515,41 @@ func getJsonSyntaxError(testJSON []byte) (bool, string) {
 	return false, ""
 }
 
+func (deps *endpointDeps) processStoredAuctionResponses(ctx context.Context, impInfo []ImpExtPrebidData) (map[string]json.RawMessage, []error) {
+	storedAuctionResponseIds := make([]string, 0, 0) //all stored responses ids from all imps
+	//because of bulk fetch responses we need to map imp id to stored resp body
+	impIdToRespId := make(map[string]string)              //imp id to stored resp id
+	impIdToStoredResp := make(map[string]json.RawMessage) //imp id to stored resp body
+	for index, impData := range impInfo {
+
+		if impData.ImpExtPrebid.StoredAuctionResponse != nil {
+			if len(impData.ImpExtPrebid.StoredAuctionResponse.ID) == 0 {
+				return nil, []error{fmt.Errorf("request.imp[%d] has ext.prebid.storedauctionresponse specified, but \"id\" field is missing ", index)}
+			}
+			storedAuctionResponseIds = append(storedAuctionResponseIds, impData.ImpExtPrebid.StoredAuctionResponse.ID)
+
+			impId, err := jsonparser.GetString(impData.Imp, "id")
+			if err != nil {
+				return nil, []error{err}
+			}
+
+			impIdToRespId[impId] = impData.ImpExtPrebid.StoredAuctionResponse.ID
+
+		}
+	}
+	if len(storedAuctionResponseIds) > 0 {
+		storedAuctionResponses, errs := deps.storedRespFetcher.FetchResponses(ctx, storedAuctionResponseIds)
+		if len(errs) > 0 {
+			return nil, errs
+		}
+		for impId, respId := range impIdToRespId {
+			impIdToStoredResp[impId] = storedAuctionResponses[respId]
+		}
+		return impIdToStoredResp, nil
+	}
+	return nil, nil
```

Comment:
To fetch stored responses  in `getAllBids` we will need to pass fetcher all the way down the stack. We don't  have this option now. 
Also `processStoredAuctionResponses` function is a part of `parseRequest` where previous function is `processStoredRequests`. To me it makes sense to have it all in one place and fetch extra data sooner (and error out in case of errors) than later. 
Fetching bids is an http calls, not fetcher calls, it would be confusing to mix this logic. In this case we don't even go that deep, we don't call `getAllBids` if there were stored responses.